customize_image.sh

```diff
@@ -85,7 +85,33 @@ cleanup() {
 	cleanup_losetup
 }
 
-trap cleanup ERR SIGINT
+build_error() {
+  cleanup
+  set +x
+  echo -e '\e[31;1m'
+  echo '====================================================================================='
+  echo '|                                                                                   |'
+  echo '| THE BUILD PROCESS WAS INTERRUPTED / FAILED. THE RESULTING IMAGE MIGHT BE UNUSABLE |'
+  echo '|                                                                                   |'
+  echo '====================================================================================='
+  echo -e '\e[0m'
+
+}
+
+build_success() {
+  cleanup
+  set +x
+  echo -e '\e[32;1m'
+  echo '==================================='
+  echo '|                                 |'
+  echo '| THE BUILD PROCESS WAS SUCCESFUL |'
+  echo '|                                 |'
+  echo '==================================='
+  echo -e '\e[0m'
+
+}
+
+trap build_error ERR SIGINT
```

Comment:
Does the script when an error happen, return an error code when exiting or is it zero?

Reply:
As the shell option -e is set above,  when error happens, the script will exit with the value returned by last command.
`-e  Exit immediately if a command exits with a non-zero status.
`
So here, both the build_error() and build_success() return the zero (from execution of "echo -e ..."), and the script will always return zero ( when the TRAP get catched).

Suggestion: add exit 0 to end of build_successfu() and exit 1 to end of build_error() for 2 purposes:
1. end the script explicitly when error happens or build finished (even no -e shell option, or it is lost).
2. give differentiate return value to the caller to indicate failure or success